Application/db/seven_queries.sql

```diff
@@ -1,60 +1,95 @@
 
--- 3. All ingredients which were imported from the Supplier "Extro AG" with the ID = 3
-SELECT a.ingredient_id, a.ingredient_name, b.supplier_id, b.supplier_name
-FROM ingredient a JOIN supplier b
-    ON a.supplier_id = b.supplier_id
-WHERE b.supplier_id = '3'
+--2. Select all customer, who ordered Kiwi Smoothie and live in München
+SELECT
+    c.f_name || ' ' || c.l_name AS customer_name,
+    c.street_name || ', ' || c.zip_code ||', ' || c.city AS address,
+	o.orders_id AS orders_id,
+	o.orders_date AS order_date,
+	p.product_name AS product
+FROM
+    customer AS c
+    LEFT JOIN orders AS o ON (c.customer_id = o.customer_id)
+    LEFT JOIN order_details AS d ON (o.orders_id = d.orders_id)
+    LEFT JOIN product AS p ON (d.product_id = d.product_id)
+WHERE
+    c.city = 'München'
+    AND p.product_name = 'Kiwi Smoothie';
 
+--3. List the import history of every ingredient and total purchase price
+SELECT
+	i.ingredient_name AS ingredient,
+	s.supplier_name as from_supplier,
+	i.quantity AS quantity,
+	i.price AS price_per_fruit,
+	(i.quantity * i.price) AS invoice_amount
+FROM
+    ingredient AS i
+    LEFT JOIN supplier AS s ON (i.supplier_id = s.supplier_id)
+    
+--4. Show the accumulated revenue of every product
+SELECT
+	p.product_id AS product_id,
+	p.product_name AS product,
+	p.selling_price,
+	SUM(d.quantity * p.selling_price) AS product_Revenue
+FROM
+    product AS p
+    LEFT JOIN order_details AS d ON (p.product_id = d.product_id)
+    LEFT JOIN orders AS o ON (d.orders_id = o.orders_id)
+GROUP BY
+    p.product_id
+ORDER BY 
+	p.product_id ASC;
 
--- 4. concatenation of strings: street_name , zip_code, city of supplier were written as adress
-SELECT supplier_id, supplier_name,
-    street_name ||', '|| zip_code ||' '|| city AS adress
-FROM supplier
+-- 5. List of the products, which contain ingredient_id = 5 or 6
+SELECT 
+	b.product_id, 
+	b.product_name AS Product,
+	c.category_name AS category,
+	d.ingredient_name AS ingredient
+FROM 
+	recipe a 
+	JOIN product b ON a.product_id = b.product_id
+	JOIN category c ON b.category_id = c.category_id 
+	JOIN ingredient d ON a.ingredient_id = d.ingredient_id
+WHERE 
+	d.ingredient_id = '5' 
+    OR 
+    d.ingredient_id = '6'
 
--- 5. How many times was the product ordered?
-SELECT DISTINCT c.product_id, COUNT(b.quantity) AS ordered_times
-FROM order_details b JOIN product c
-    ON b.product_id = c.product_id
-GROUP BY c.product_id
-
--- 6. List of the products, which contain ingredient_id = 5
-SELECT a.product_id, a.product_name
-FROM product a JOIN category b
-    ON a.product_id = b.product_id 
-JOIN recipe c
-    ON a.product_id = c.product_id
-JOIN ingredient d
-    ON c.ingredient_id = d.ingredient_id
-WHERE d.ingredient_id = '5'
+--6. Lists all orders made in the last seven days of January
+SELECT *
+FROM orders
+WHERE
+    orders_date BETWEEN DATE '2022-01-01'
+    AND DATE '2022-01-31'
+EXCEPT
+SELECT *
+FROM orders
+WHERE
+    orders_date <= DATE '2021-01-25';
```

Comment:
This one doesn't seem right. There are 2021 and 2022

Reply:
My bad :) it should be 2022